webapp/src/index.tsx

```diff
@@ -416,14 +417,14 @@ export default class Plugin {
             const roles = getMyRoles(store.getState());
             const cms = getMyChannelMemberships(store.getState());
 
-            registry.unregisterComponent(channelHeaderMenuID);
-
             try {
                 const resp = await axios.get(`${getPluginPath()}/config`);
+                registry.unregisterComponent(channelHeaderMenuID);
```

Comment:
I'm not really sure what this change does... We're putting the `unregisterComponent` call in the `try`, just incase the `unregisterComponent` throws? But the `unregister` is a `dispatch` call, which can't throw (unless we use middleware that does?) -- so I'm not really sure if this change does anything? 
But anyway, I'll approve because it doesn't really hurt I guess. :)

Reply:
It's a bit tricky but I'll try to explain. This line change is the fix for:

![image](https://user-images.githubusercontent.com/1832946/155674507-911affb2-9aa8-44f0-9fe3-2b58c3f1e3fa.png)

We need to do the unregistration after the async call that fetches the config otherwise it will risk of running multiple times (e.g. quickly switching channels) before we get a chance to register again down below which means we'll "lose" unregister calls. Then the register calls will all happen once those queued async calls finish which will cause a leak given the unregisters already happened with stale ids.

This is a sample sequence of the bug happening:

- Channel switch 1
  - unregistering (this is a no-op as there's nothing registered yet, `channelHeaderMenuID` is empty)
  - registered ID A
- Channel switch 2:
  - unregistering ID A
  - async call taking some time
  - Channel switch 3
    - unregistering ID A again (wasting a call)
    - async call
    - async call returned
    - registering ID C
  - async call returned
  - registering ID B

Reply:
Ah, gotcha. Very clear explanation, thank you!